subsys/net/lib/openthread/platform/logging.c

```diff
@@ -19,6 +19,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #define LOG_PARSE_BUFFER_SIZE  128
 
+#if !CONFIG_LOG_BACKEND_SPINEL
```

Comment:
I think it would be cleaner to disable `logging.c` build at the CMake level (make in conditional with `zephyr_library_sources_ifndef`), there's really no point building this file in this case.

Reply:
Very smart way. I find it really useful to remember :) Done